algorithms/kernel/dbscan/oneapi/dbscan_dense_default_batch_ucapi_impl.i

```diff
@@ -202,7 +203,14 @@ Status DBSCANBatchKernelUCAPI<algorithmFPType>::compute(const NumericTable * x,
     DAAL_CHECK_STATUS_VAR(s);
 
     DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(uint32_t, _queueBlockSize, nRows);
-    DAAL_CHECK_STATUS_VAR(initializeBuffers(nRows));
+    uint32_t chunkSize   = _recommendedChunkSize;
+    uint32_t chunkNumber = nRows / chunkSize + uint32_t(bool(nRows % chunkSize));
+    while (chunkNumber < _minRecommendedNumberOfChunks && chunkSize > _minChunkSize)
+    {
+        chunkSize /= 2;
+        chunkNumber = nRows / chunkSize + uint32_t(bool(nRows % chunkSize));
+    }
```

Comment:
You can extract method `calculateChunkNumber()` from this code, make `chunkNumber` variable a field of `DBSCANBatchKernelUCAPI` object. In this case, I would recommend:
- remove an additional parameter `chunkNumber` from several functions like `initializeBuffers`.
- initialize `_chunkNumber` in the constructor with minimum/recommended value. It will be recalculated in the `compute()` method.